app/serializers/search/goods_nomenclature_serializer.rb

```diff
@@ -7,7 +7,7 @@ def serializable_hash(_opts = {})
         serializable = {
           id:,
           goods_nomenclature_item_id:,
-          short_code: short_code(fetch_commodity_short_code: true),
+          short_code: respond_to?(:specific_system_short_code) ? specific_system_short_code : short_code,
```

Comment:
This teaches the generic serializer about varients of GoodsNomenclature

Whats the scenario where system specific short code isn't available? Can you just add an implementation of this to GoodsNomenclature, and have that call `#short_code`. Then it would be overridden by TenDigit ?

Then all goods_nomenclatures behave the same and the serializer does need logic to determine how to serializer the GN?

Reply:
That's a good shout, actually.